core/domain/collection_domain.py

```diff
@@ -410,6 +526,47 @@ def _convert_collection_contents_v2_dict_to_v3_dict(
         return collection_contents
 
     @classmethod
+    def _convert_collection_contents_v3_dict_to_v4_dict(
+            cls, collection_contents):
+        """Converts from version 3 to 4.
+
+        Adds a skills dict and skill id counter. Gets skills in
+        prerequisite_skills and acquired_skills in nodes, and assigns them
+        integer IDs.
+        """
+
+        skill_names = set()
+        for node in collection_contents['nodes']:
+            skill_names.update(node['acquired_skills'])
+            skill_names.update(node['prerequisite_skills'])
+        skill_names_to_ids = {
+            name: 's' + str(index)
+            for index, name in enumerate(sorted(skill_names))
+        }
+
+        collection_contents['nodes'] = [{
+            'exploration_id': node['exploration_id'],
+            'prerequisite_skills': [
```

Comment:
Should we rename these fields to ``prerequisite_skill_ids`` and ``acquired_skill_ids``?